### PR TITLE
Update etcd 3.5.5 and all otel related libraries to newer versions

### LINKED ISSUE #106536: Use of outdated OpenTelemetry Go package
### What happened?

Importing Kubernetes' `k8s.io/apiserver` module together with the latest version of `go.opentelemetry.io/otel` results in a failed build due to breaking changes in the OpenTelemetry Go library. This also happens with `k8s.io/component-base`.

The solution is to bump the opentelemetry dependencies to a stable version.

### What did you expect to happen?

`k8s.io/apiserver` should be usable in a module that uses a stable version of `go.opentelemetry.io/otel` 

### How can we reproduce it (as minimally and precisely as possible)?

A simple way to reproduce it is to create a new Go module via `go mod init example.com/kube-issue` that depends on the latest version of both `k8s.io/apiserver` and some new `go.opentelemetry.io/otel` module, e.g. `go.opentelemetry.io/otel/exporters/otlptrace/otlptracegrpc`.

<details>
<summary>main.go contents</summary>

```go
package main

import (
	_ "go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
	_ "k8s.io/apiserver/pkg/endpoints/filters"
)

```

</details>

If you try to do `go mod tidy` you can see a bunch of errors (which come from breaking changes on the OpenTelemetry Go libraries).

<details>

<summary> <code>go mod tidy</code> output</summary>

```
❯ go mod tidy
go: finding module for package go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc
go: downloading go.opentelemetry.io/otel/exporters/otlp/otlptrace v1.2.0
go: downloading go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.2.0
go: found go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc in go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc v1.2.0
go: downloading go.opentelemetry.io/proto/otlp v0.10.0
go: finding module for package go.opentelemetry.io/otel/semconv
go: finding module for package go.opentelemetry.io/otel/unit
go: finding module for package go.opentelemetry.io/otel/internal/metric
go: found go.opentelemetry.io/otel/internal/metric in go.opentelemetry.io/otel/internal/metric v0.25.0
go: finding module for package go.opentelemetry.io/otel/metric/registry
go: finding module for package go.opentelemetry.io/otel/semconv
example.com/kube-issue imports
        k8s.io/apiserver/pkg/endpoints/filters imports
        go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp imports
        go.opentelemetry.io/otel/semconv: module go.opentelemetry.io/otel@latest found (v1.2.0), but does not contain package go.opentelemetry.io/otel/semconv
example.com/kube-issue imports
        k8s.io/apiserver/pkg/endpoints/filters imports
        go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp tested by
        go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp.test imports
        go.opentelemetry.io/otel/oteltest imports
        go.opentelemetry.io/otel/metric/registry: module go.opentelemetry.io/otel/metric@latest found (v0.25.0), but does not contain package go.opentelemetry.io/otel/metric/registry
```
</details> 

### Anything else we need to know?

Probably @dashpole is the right person to address this (but I am happy to help with it!)

### Kubernetes version

Any version that includes #94942, that is, `v0.22.0` or later.

### Cloud provider

n/a

### OS version

n/a

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by dims**
We have updated the etcd server we use in `master` to v3.5.5. In this PR, we update the vendored dependencies. 

Note that to update etcd dependency, we need to update to newer opentelemetry libraries and API.

#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes:

Fixes #106536

#### Does this PR introduce a user-facing change?

```release-note
API Server tracing root span name for opentelemetry is changed from "KubernetesAPI" to "HTTP GET"
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/go.mod [hunk 1]
```diff
@@ -4,17 +4,19 @@ module k8s.io/api
 
 go 1.19
 
+exclude go.opentelemetry.io/contrib/propagators v0.20.0
```
Comment (dims) [inline staging/src/k8s.io/api/go.mod:7]: We have our very first `exclude` statement!!!

Comment (dims) [reply staging/src/k8s.io/api/go.mod]: let's see if things work without it.

**Comment by dims** [comment]
@sallyom @hanyuancheung i got as far as getting all the CI jobs running except for `TestAPIServerTracing`. I may be missing something in the new API for the various opentelemetry libraries. Can you please help fix this? (see example failure in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/112545/pull-kubernetes-integration/1571283051768451072)

**Comment by dims** [comment]
@Verolop integration test will not go green yet :) i've asked a few people to help diagnosein https://github.com/kubernetes/kubernetes/pull/112545#issuecomment-1250172298

File: cmd/kubelet/app/server.go [hunk 2]
```diff
@@ -39,9 +39,9 @@ import (
 	"k8s.io/mount-utils"
 
 	cadvisorapi "github.com/google/cadvisor/info/v1"
-	"go.opentelemetry.io/otel/exporters/otlp/otlpgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
 	otelsdkresource "go.opentelemetry.io/otel/sdk/resource"
-	"go.opentelemetry.io/otel/semconv"
+	"go.opentelemetry.io/otel/semconv/v1.10.0"
```
Comment (ash2k) [inline cmd/kubelet/app/server.go:44]: `v1.12.0` is the latest sem version. I wonder why not use it? https://pkg.go.dev/go.opentelemetry.io/otel/semconv

Comment (dims) [reply cmd/kubelet/app/server.go]: Done!

**Comment by hanyuancheung** [comment]
> @sallyom @hanyuancheung i got as far as getting all the CI jobs running except for `TestAPIServerTracing`. I may be missing something in the new API for the various opentelemetry libraries. Can you please help fix this? (see example failure in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/112545/pull-kubernetes-integration/1571283051768451072)

@dims Sorry for missing this msg. Now version of otlptracegrpc from OTEL looks good to me.

**Review by hanyuancheung — APPROVED** [review]

**Comment by dashpole** [comment]
Thanks for taking this on @dims!  OpenTelemetry-related changes LGTM
/approve

File: staging/src/k8s.io/apiserver/pkg/server/options/tracing.go [hunk 3]
```diff
@@ -23,9 +23,9 @@ import (
 	"net"
 
 	"github.com/spf13/pflag"
-	"go.opentelemetry.io/otel/exporters/otlp/otlpgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
 	"go.opentelemetry.io/otel/sdk/resource"
-	"go.opentelemetry.io/otel/semconv"
+	"go.opentelemetry.io/otel/semconv/v1.12.0"
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/server/options/tracing.go:28]: @dashpole are there any user-facing impacts of this version change we need to release note?

File: test/integration/apiserver/tracing/tracing_test.go [hunk 4]
```diff
@@ -107,15 +108,18 @@ endpoint: %s`, listener.Addr().String())), os.FileMode(0755)); err != nil {
 
 func containsNodeListSpan(req *traceservice.ExportTraceServiceRequest) bool {
 	for _, resourceSpans := range req.GetResourceSpans() {
-		for _, instrumentationSpans := range resourceSpans.GetInstrumentationLibrarySpans() {
+		for _, instrumentationSpans := range resourceSpans.GetScopeSpans() {
 			for _, span := range instrumentationSpans.GetSpans() {
-				if span.Name != "KubernetesAPI" {
+				if span.Name != "HTTP GET" {
```
Comment (liggitt) [inline test/integration/apiserver/tracing/tracing_test.go:113]: @dashpole is this a test-only change, or an example of a user-facing change in span names?

File: go.mod [hunk 5]
```diff
@@ -89,7 +89,7 @@ require (
 	gonum.org/v1/gonum v0.6.2
 	google.golang.org/api v0.60.0
 	google.golang.org/genproto v0.0.0-20220502173005-c8bf987b8c21
-	google.golang.org/grpc v1.47.0
+	google.golang.org/grpc v1.49.0
```
Comment (liggitt) [inline go.mod:92]: curious what prompted this change, etcd 3.5.5 is still on 1.41.0 (https://github.com/etcd-io/etcd/blob/release-3.5/go.mod#L35)

nothing breaking-looking jumps out at me from https://github.com/grpc/grpc-go/releases, but the diff isn't very reviewable, so just wondering

File: staging/src/k8s.io/apiserver/pkg/server/options/tracing.go (hunk 3 above)
Comment (dashpole) [reply staging/src/k8s.io/apiserver/pkg/server/options/tracing.go:28]: This is not user-facing.  The conventions we use (just service.name) do not differ between versions.

File: test/integration/apiserver/tracing/tracing_test.go (hunk 4 above)
Comment (dashpole) [reply test/integration/apiserver/tracing/tracing_test.go:113]: This is a user-facing change in span names.  There will likely be another change after https://github.com/kubernetes/kubernetes/issues/112059 is addressed.  We can call it out.

Comment (liggitt) [reply test/integration/apiserver/tracing/tracing_test.go:113]: ack, a change there is in-bounds given the feature is alpha. Can you give @dims language to use in the release note?

File: LICENSES/vendor/github.com/grpc-ecosystem/grpc-gateway/v2/LICENSE [hunk 6]
```diff
@@ -0,0 +1,31 @@
+= vendor/github.com/grpc-ecosystem/grpc-gateway/v2 licensed under: =
```
[the remark this answers is not in the document]
Comment (dims) [reply LICENSES/vendor/github.com/grpc-ecosystem/grpc-gateway/v2/LICENSE:1]: good point will open one

File: test/integration/apiserver/tracing/tracing_test.go (hunk 4 above)
Comment (dims) [reply test/integration/apiserver/tracing/tracing_test.go:113]: updated release note

File: go.mod (hunk 5 above)
Comment (dims) [reply go.mod:92]: i did not do this explicitly, it may have some from one of the otel libraries i think.

Comment (liggitt) [reply go.mod:92]: Mm, ok.

Comment (dims) [reply go.mod:92]: if i force to pin the dependency on the old version (`1.47.0`) here's what happens.

```
[dims@dims-m1-7728 17:56] ~/go/src/k8s.io/kubernetes ⟩ hack/pin-dependency.sh google.golang.org/grpc v1.47.0
Running: go mod download google.golang.org/grpc@v1.47.0
Resolved to google.golang.org/grpc@v1.47.0
Running: go mod edit -require google.golang.org/grpc@v1.47.0
Running: go mod edit -replace google.golang.org/grpc=google.golang.org/grpc@v1.47.0

Run hack/update-vendor.sh to rebuild the vendor directory


[dims@dims-m1-7728 17:56] ~/go/src/k8s.io/kubernetes ⟩ hack/update-vendor.sh
+++ [0919 17:56:49] logfile at /tmp/update-vendor.Py7b/update-vendor.log
+++ [0919 17:56:49] go.mod: update staging references
+++ [0919 17:56:57] go.mod: propagate to staging modules
+++ [0919 17:56:57] go.mod: sorting staging modules
+++ [0919 17:57:12] go.mod: tidying
+++ [0919 17:57:23] go.mod: adding generated comments
+++ [0919 17:57:23] vendor: updating internal modules
+++ [0919 17:57:23] vendor: running 'go mod vendor'
+++ [0919 17:57:27] vendor: updating vendor/LICENSES
+++ [0919 17:57:33] vendor: creating OWNERS file
+++ [0919 17:57:33] NOTE: don't forget to handle vendor/* files that were added or removed


[dims@dims-m1-7728 17:57] ~/go/src/k8s.io/kubernetes ⟩ hack/lint-dependencies.sh
Check all unwanted dependencies and update its status.
These modules are pinned to versions different than the minimal preferred version.
That means that without replace directives, a different version would be selected,
which breaks consumers of our published modules.
1. Use hack/pin-dependency.sh to switch to the preferred version for each module
2. Run hack/update-vendor.sh to rebuild the vendor directory
3. Run hack/lint-dependencies.sh to verify no additional changes are required

google.golang.org/grpc
    pinned:    v1.47.0
    preferred: v1.49.0
    hack/pin-dependency.sh google.golang.org/grpc v1.49.0
All pinned versions of checked dependencies match their preferred version.
```

Comment (dims) [reply go.mod:92]: it's coming from here:
https://github.com/open-telemetry/opentelemetry-go-contrib/blob/instrumentation/google.golang.org/grpc/otelgrpc/v0.35.0/instrumentation/google.golang.org/grpc/otelgrpc/go.mod#L9

`go mod graph` shows:
```
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.35.0 google.golang.org/grpc@v1.49.0
```

File: LICENSES/vendor/github.com/grpc-ecosystem/grpc-gateway/v2/LICENSE (hunk 6 above)
[the remark this answers is not in the document]
Comment (dims) [reply LICENSES/vendor/github.com/grpc-ecosystem/grpc-gateway/v2/LICENSE:1]: hmm, it's coming from the otel libraries as well

go mod graph shows:
```
go.opentelemetry.io/otel/exporters/otlp/otlptrace@v1.10.0 github.com/grpc-ecosystem/grpc-gateway/v2@v2.7.0
go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc@v1.10.0 github.com/grpc-ecosystem/grpc-gateway/v2@v2.7.0
go.opentelemetry.io/proto/otlp@v0.19.0 github.com/grpc-ecosystem/grpc-gateway/v2@v2.7.0
```

**Pull request merged**